src/vg/civcraft/mc/civmenu/TOSListener.java

```diff
@@ -66,13 +65,27 @@ public void run() {
 	public void playerMoveEvent(PlayerMoveEvent event) {
 		Player p = event.getPlayer();
 		if (!TOSManager.isTermPlayer(p, "CivMenu Agreement")) {
-			if(event.getTo().distance(locations.get(p.getUniqueId())) > plugin.GetConfig().get("terms.MovementRange").getInt()){
+			Location from = locations.get(p.getUniqueId());
+			if (from == null){
+				from = event.getFrom();
+				locations.put(p.getUniqueId(), from);
```

Comment:
I doubt you can slip through the login listener. The problem is more likely be something with the sql manager returning false even though the player has already registered and was removed from the map.